server.py

```diff
@@ -911,6 +1001,79 @@ def do_POST(self):
             print(google_email.get_email_history(email_history['historyId']))
             self.send_204()
 
+        elif self.path == "/getObservationStations":
+            try:
+                user_jwt = json_body["jwt"]
+                decoded_jwt = decode_jwt(user_jwt)
+                jwt_user_addr = decoded_jwt["address"]
+            except Exception as e:
+                print(e)
+                pass
+            if isValidEthereumAddress(jwt_user_addr) is False:
+                self.send_400(message='Invalid Ethereum address', explain='Ethereum address pulled from user JWT is not valid')
+                return
+            try:
+                observation_station_numbers = self.db.selectUserStationNumbers_JSON(jwt_user_addr)
+                self.send_200_JSON(json.dumps(observation_station_numbers))
+                return
+            except Exception as e:
+                print(e)
+                self.send_500(message='Could not get station information', explain='Query to get observation stations has failed')
+
+        elif self.path == '/generateStation':
+            try:
+                user_jwt = json_body["jwt"]
+                decoded_jwt = decode_jwt(user_jwt)
+                user_addr = decoded_jwt["address"]
+                station_name = json_body["station"]
+                latitude = json_body["latitude"]
+                longitude = json_body["longitude"]
+                elevation = json_body["elevation"]
+                notes = json_body["notes"]
+            except Exception as e:
+                print(e)
+                self.send_400(message='Missing parameter(s)', explain='One or more of the required parameters are missing form the request body')
+                return
+            if isValidEthereumAddress(user_addr) is False:
+                self.send_400(message='message', explain='explanation')
+                return
+            try:
+                user_id = self.db.selectObserverIDFromAddress(user_addr)
+                latest_station = self.db.selectLatestStationID()
+                if latest_station[0:1] == 'T':
+                    station_index = latest_station[1:]
+                    station_index = int(station_index, 36) + 1
+                    # Prevent I or O character
+                    if (station_index & 18) == 18 or (station_index & 24) == 24:
+                        station_index = station_index + 1
+                    if (station_index & 648) == 648 or (station_index & 864) == 864:
+                        station_index = station_index + 36
+                    if (station_index & 23328) == 23328 or (station_index & 31104) == 31104:
+                        station_index = station_index + 1296
```

Comment:
This doesn't look like it would prevent I and Os in all cases - Just TI00 T0I0 and T00I, or TO00, T0O0 or T00O.  What about cases like TH1H (which appears to fail)?

Reply:
I double checked, `&` is the bitwise operator for `AND` which will isolate each index. I am not sure if this was confused with `&&` which is the logical operator but I can also be missing something.